make/go.mk

```diff
@@ -21,8 +21,21 @@ $(OUT_DIR)/operator:
 vendor:
 	$(Q)go mod vendor
 
+NSTEMPLATES_DIR=deploy/nstemplatetiers
+YAML_TEMPLATES := $(wildcard $(NSTEMPLATES_DIR)/*-*.yaml)
+
 .PHONY: generate
-generate:
+generate: generate-metadata
 	@echo "generating templates bindata..."
 	@go install github.com/go-bindata/go-bindata/...
-	@$(GOPATH)/bin/go-bindata -pkg templates -o ./pkg/templates/template_contents.go -nocompress -prefix deploy/nstemplatetiers deploy/nstemplatetiers
\ No newline at end of file
+	@$(GOPATH)/bin/go-bindata -pkg templates -o ./pkg/templates/templates.go -nocompress -prefix $(NSTEMPLATES_DIR) $(NSTEMPLATES_DIR)
+
+.PHONY: generate-metadata
+generate-metadata:
+	@echo "generating namespace templates metadata in $(YAML_TEMPLATES)" 
+	@-rm $(NSTEMPLATES_DIR)/metadata.yaml
```

Comment:
can we silently ignore when `metadata.yaml` is  absent, I can see
```
rm: cannot remove 'deploy/nstemplatetiers/metadata.yaml': No such file or directory
make: [make/go.mk:36: generate-metadata] Error 1 (ignored)
```

Reply:
yes: if the command starts with a `-`, the error is ignored. (see https://www2.ipp.mpg.de/~dpc/gmake/make_47.html)

Reply:
I think that @dipak-pawar meant ignoring also the error message: 
```
@-rm $(NSTEMPLATES_DIR)/metadata.yaml 2> /dev/null
```
but that's a detail

Reply:
ah sorry, I misunderstood @dipak-pawar 's comment. That's a good idea, actually! I'll add that in a subsequent PR ;)